onnxruntime/contrib_ops/cpu/transformers/generate_impl_base.h

```diff
@@ -33,6 +33,40 @@ gsl::span<T> AllocateBuffer(AllocatorPtr allocator,
   return span;
 }
 
+template <typename ElementType>
+inline void AllocateTempBufferForGetGreedySearchTopOne(
+    int32_t batch_size,
+    AllocatorPtr allocator,
+    BufferUniquePtr& buffer,
+    gsl::span<ElementType>& stage_1_scores,  // shape (batch_size, parts_of_vocab)
+    gsl::span<int32_t>& stage_1_tokens,      // shape (batch_size, parts_of_vocab)
+    gsl::span<ElementType>& output_scores,   // shape (batch_size)
+    gsl::span<int32_t>& output_tokens        // shape (batch_size)
+) {
+  constexpr size_t kMaxPartsPerVocab = 128;
+  const size_t stage_1_element_size = kMaxPartsPerVocab * batch_size;
+  const size_t output_element_size = batch_size;
+
+  // Note: use float to allocate buffer for temporary value buffer to avoid unalignment
```

Comment:
If Alloc api will get unalignment memory, we shall fix Alloc itself...

Reply:
It is not caused by Alloc. One continuous buffer is allocated for stage1_score, stage1_token, stage2_score, stage2_token. If stage_1_element_size is odd and type is fp16, then buffer size for stage1_score may not a multiply of 4. Howerver, stage1_token (int32_t) requires alignment of 4. For simplicity, I just use float to allocate the temp buffer as it is small.